federation/normalize.go

```diff
@@ -171,18 +171,18 @@ func mergeSameAlias(selections []*graphql.Selection) ([]*graphql.Selection, erro
 				return nil, fmt.Errorf("one selection with alias %s has subselections and one does not",
 					selection.Alias)
 			}
-			seenSelections := make(map[string]bool, len(last.SelectionSet.Selections))
+			seenSelections := make(map[string]struct{}, len(selection.SelectionSet.Selections))
 			for _, s := range selection.SelectionSet.Selections {
-				if !seenSelections[s.Alias] {
-					seenSelections[s.Alias] = true
+				if _, ok := seenSelections[s.Alias]; !ok {
+					seenSelections[s.Alias] = struct{}{}
 					last.SelectionSet.Selections = append(last.SelectionSet.Selections, s)
 				}
 
 			}
-			seenFragments := make(map[*graphql.Fragment]bool, len(last.SelectionSet.Selections))
+			seenFragments := make(map[*graphql.Fragment]struct{}, len(selection.SelectionSet.Fragments))
```

Comment:
In the commit message also call out that this corrects the initial size to be based on Fragments rather than Selections.
I expect that there are generally many more Selections than Fragments, so I think this means that this allocated space that would go unused, but it didn't cause additional allocations.